tests/integration/test_nonzero.py

```diff
@@ -75,36 +91,44 @@ def test():
             [[3, 0], [0, 3], [0, 0], [2, 2], [0, 19]],
         ]
     )
+
+    assert_equal(num.nonzero(x), np.nonzero(x_np))
     assert num.count_nonzero(x) == np.count_nonzero(x_np)
     assert num.count_nonzero(x, axis=(0, 1, 2)) == np.count_nonzero(
         x_np, axis=(0, 1, 2)
     )
+
     # TODO: Put this back once we have per-axis count_nonzero
     # for axis in range(3):
     #    assert_equal(
     #        num.count_nonzero(x, axis=axis),
     #        np.count_nonzero(x_np, axis=axis),
     #    )
-    assert_equal(num.nonzero(x), np.nonzero(x_np))
 
-    assert_equal(num.nonzero(0), ([],))
-    assert_equal(num.nonzero(1), ([0],))
 
-    x_np = np.random.randn(100)
-    indices = np.random.choice(
-        np.arange(x_np.size), replace=False, size=int(x_np.size * 0.2)
-    )
-    x_np[indices] = 0
-    x = num.array(x_np)
-    assert num.count_nonzero(x) == np.count_nonzero(x_np)
-    lg_nonzero = num.nonzero(x)
-    np_nonzero = np.nonzero(x_np)
-    assert_equal(lg_nonzero, np_nonzero)
+def test_deprecated_0d():
+    with pytest.deprecated_call():
+        assert num.count_nonzero(num.array(0)) == 0
+        assert num.count_nonzero(num.array(0, dtype="?")) == 0
+        assert_equal(num.nonzero(0), np.nonzero(0))
+
+    with pytest.deprecated_call():
+        assert num.count_nonzero(num.array(1)) == 1
+        assert num.count_nonzero(num.array(1, dtype="?")) == 1
+        assert_equal(num.nonzero(1), np.nonzero(1))
+
+    with pytest.deprecated_call():
+        assert_equal(num.nonzero(0), ([],))
+
+    with pytest.deprecated_call():
+        assert_equal(num.nonzero(1), ([0],))
```

Comment:
Here is an example of using `pytest.deprecated_call()`. This will ensure the annoying deprecation messages do not clutter the test report, but also more importantly, maintains the fact of the deprecations themselves under test.